docs/tutorials/embedding/MASE.ipynb

```diff
@@ -0,0 +1,377 @@
+   "metadata": {},
+   "source": [
+    "We sample two 2-block SBMs (undirected, no self-loops) with 50 vertices, each block containing 25 vertices ($n = [25, 25]$), and with the following block probabilities:\n",
+    "\n",
+    "\\begin{align*}\n",
+    "P_1 = \n",
+    "\\begin{bmatrix}0.3 & 0.1\\\\\n",
+    "0.1 & 0.7\n",
+    "\\end{bmatrix},~\n",
+    "P_2 = \\begin{bmatrix}0.3 & 0.1\\\\\n",
+    "0.1 & 0.3\n",
+    "\\end{bmatrix}\n",
+    "\\end{align*}\n",
+    "\n",
+    "The only difference between the two is the within-block probability for the second block. We sample $G_1 \\sim \\text{SBM}(n, P_1)$ and $G_2 \\sim \\text{SBM}(n, P_2)$."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from graspologic.simulations import sbm\n",
+    "\n",
+    "n = [25, 25]\n",
+    "P1 = [[0.3, 0.1],\n",
+    "      [0.1, 0.7]]\n",
+    "P2 = [[0.3, 0.1],\n",
+    "      [0.1, 0.3]]\n",
+    "\n",
+    "np.random.seed(8)\n",
+    "G1 = sbm(n, P1)\n",
+    "G2 = sbm(n, P2)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Visualize the graphs using heatmap\n",
+    "\n",
+    "We visualize the sampled graphs using the ``heatmap()`` function. ``heatmap()`` will plot the adjacency matrix, where the colors represent the weight of each edge. In this case, we have binary graphs so the values will be either 0 or 1. \n",
+    "\n",
+    "We see that there is clear block structure to the graphs. Furthermore, the lower right quarter of $G_1$, representing the within-group connections for the second group, is more dense than that of $G_2$."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from graspologic.plot import heatmap\n",
+    "\n",
+    "g1 = heatmap(G1, figsize=(7,7), title=\"Visualization of Graph 1\")\n",
+    "g2 = heatmap(G2, figsize=(7,7), title=\"Visualization of Graph 2\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Embed the two graphs using MASE"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Just as ASE fits a single input graph to the Random Dot Product Graph (RDPG) model, MASE fits a sample of graphs to the common subspace independent-edge (COSIE) model, which is an extension of RDPG for multiple graphs. The COSIE model assumes that all of the graphs have important shared properties, so all graphs are derived from a common subspace with orthonormal basis $V$.  But, COSIE also allows the connection probabilities between each vertex to vary from one graph to the next, so each graph $A^{(i)}$ has a corresponding score matrix $R^{(i)}$.  As a result, COSIE facilitates comparison due to the common subspace, while retaining sufficient flexibility to represent heterogeneous collections of graphs, due to the different score matrices."
```

Comment:
this is probably my most important comment of this review: after parsing through this several times, i'm starting to understand what mase does, so the tutorial is less confusing to me compared to how it was. that said, i think that it's mostly my cognitive bias, and the tutorial can (and i'd say should) be made in simpler terms than subspaces, orthonormal basis. etc. i.e. say that a a vertex j in a graph has a position that is shared between each graphs, and then each vertex/dimension also has a score, etc.-etc. (most people who use this haven't taken 2 linear algebra classes). furthermore, please specify the dimensions of matrices that you are using. because i initially thought each vertex has its own score, but the only way matrix multiplication works out is if R^i is in d by d.

tl;dr: this still feels like a tutorial for people who already know what's going on math-wise, whereas this should be a tutorial for people who don't.

cc: @bdpedigo, @j1c

Reply:
1) I hesitate to claim that each vertex has a shared position across all the graphs because it can be easily conflated with latent positions a la RPDG.  But the entries in V alone don't really represent latent positions in that sense (eg. they dont satisfy x_i^T x_j = P_ij).  It's only when they are scaled by the score matrices that they can represent latent positions.  So, it really is a position in a common subspace.
2) I feel that I need to state what V and R represent since they are the core outputs of the MASE algorithm.

Reply:
Nevertheless, I will try to incorporate some of the concepts you mentioned.  How about this version?

Just as ASE fits a single input graph to the Random Dot Product Graph (RDPG) model, MASE fits a sample of graphs to the common subspace independent-edge (COSIE) model, which is an extension of the RDPG model to multiple graphs. The COSIE model assumes that all of the graphs have important shared properties, but also meaningful differences. As a result, each vertex has a single position in the common subspace that is invariant across all of the graphs, represented by  𝑉∈ℝ𝑛×𝑑  (where  𝑛  is the number of vertices and  𝑑  is the embedding dimension). But, each graph  𝐴(𝑖)  has a potentially different score matrix  𝑅(𝑖)∈ℝ𝑑×𝑑 , allowing the actual connection probabilities between each vertex to vary from one graph to the next. Due to this structure, COSIE facilitates comparison due to the common subspace, while retaining sufficient flexibility to represent heterogeneous collections of graphs, due to the different score matrices.